### DISCUSSION TITLE
スパムへの対応について

### DISCUSSION THREAD

**Discussion by kskwtnk** [increments/qiita-discussions#81] [アナウンス]
直近では以下のDiscussionでも個別に回答していましたが、スパムへの対応について現状のご共有をいたします！

https://github.com/increments/qiita-discussions/discussions/75
https://github.com/increments/qiita-discussions/discussions/76

## スパム全般について

大前提、運営としては少しでもスパムを減らしたいと考えています
記事を読む人、書く人、双方への体験を良くするためにも今後も取り組んでいきます！

ただし完全になくすとか、いつまでにどこまで減らすといった確約ができず……まずは現状をしっかり共有するのが誠実な運営かと思いこのDiscussionを立てました

## スパムユーザーについて

現状行っている対策は以下のようになっています

- フォローアクションをとったユーザーのデータを用いたヒューリスティックなスパム判定
- 過去のスパムユーザーのデータを用いたスパム判定
  - ハックされてしまうとよろしくないので、ここでは伏せさせてください 🙇

ただし、スパマーの手口もどんどん巧妙になっているため全てを防ぎ切れてはいません
スパム行為をする意義がなくなるようにシフトしていきたいのですが、完全な愉快犯の可能性もあるので運営も悩んでいます

## スパムカレンダーについて

去年までも多少は存在していたのですが、ここまでの数で発生したのは今年が初めてです
スパムユーザーへの対応と比べてフローが整っておらず、みなさまへご心配をおかけしてしまいました

スパムユーザーへの対応を強化していった結果、もしかしたら次の穴としてカレンダーが狙われてしまったのかもしれません

適宜削除のオペレーションを実施することは決まっているため減少はしていくと思いますが、タイミングによってはまた多くのスパムカレンダーが生まれてしまうかもしれず……ご了承いただけますと幸いです

## 最後に

他のDiscussionでいただいていたように、スパムへの対策としてこういう仕組みがあったらどうか？と提案いただけるのは非常に助かります！
性質上、詳細に「○○を導入しました」とはご報告できないかもしれませんが、取り入れさせていただきたいなと考えています

今後ともQiitaをよろしくお願いします！

**Comment by NurseAngel** [comment]
・一部の信頼のおけるユーザ(スパム報告を普段から行っている等)に対して非表示権限を与える(削除ではない)。
・非表示後、実際に削除するかの最終的権限を持つのはいままでどおりQiitaのみ。
・もちろん変なことやったりしたら権限剥奪。

とか。

最後は人の目に頼らざるを得ないのであれば、最初から人の目も入れておけばいいんじゃね的な。

あ、私は権限要らないです。

**Comment by kskwtnk** [comment]
@NurseAngel 

ご確認ありがとうございます！

たしかに、スパム報告を積極的に行ってくださっている方はいらっしゃるので、その方達が協力してくださったら非常にありがたいです！
フローの制定とか全体へのアナウンスなどもあるのですぐに実施はできなさそうですが1度議論にあげてみます！

**Reply by kskwtnk** [reply]
@NurseAngel 
こちらの件について社内で議論を実施したのですが、ユーザーの皆さまに権限を付与して代理で実行していただくのは実施しなさそうです 🙏 

今回のアドベントカレンダーでは対応がそもそもできていなかったのですが、普段のスパムユーザーに対してはシステム的な対応ができており、大多数はそちらで解消できています
来年のアドベントカレンダーではこちらの仕組みを使ってスパムカレンダーが生まれない or 生まれてもすぐに対応できるようにしていく予定です！

**Comment by Zuishin** [comment]
https://qiita.com/mod_poppo/items/a6317294ba1e39b1c3b3

こちらの他人の記事に無関係な自分の記事をいくつも貼っているユーザーがいますが、通報してもメンションしても対処されません。
貼られているものはコメントを補強するものでもなく、本当に自分の記事の SEO のためのただの広告です。

これが放置されているのは、このような使い方が問題ないからということでしょうか？
スパムについての線引きがどこにあるのかわかりませんが、もしこれが問題ないというのであれば、利用させていただこうと思います。
いかがでしょうか

**Reply by kskwtnk** [reply]
@Zuishin 
ご意見ありがとうございます！

> 通報してもメンションしても対処されません。

> スパムについての線引きがどこにあるのかわかりませんが

こちらの件ですが、記事の通報と同様に対応しております
通報がある程度の件数を超えた場合にシステム的に処理を走らせておりますので、ご了承いただけますと幸いです 🙇

**Reply by Zuishin** [reply]
なるほど、運営判断ではなく多数票によるものということですね？

今まで謎に思っていた、特定の組織に肩入れしているように見える基準が、組織票によるものとわかってすっきりしました。
ありがとうございます。

**Reply by Zuishin** [reply]
と、思ったけど、次の投稿の対応を見る限り、やはり通報の数ではなく運営判断ですね。

他人の記事へのコメントで、自分の外部サイトの記事を宣伝するのは、「する人によってはあり」ということ、了解しました。

**Comment by yoshinoyuya** [comment]
スパム対応ありがとうございます。
スパムコメントに関して情報共有をします。

記事対して、感謝を述べる英文のあと、記事に関係のないURLを貼っている、スパムコメントが多く見受けられます。
機械的に複数の投稿があるわけでは、ありませんが
感謝の英文やアカウントを変更して行っているなど
悪意のある行動だと思います。

https://qiita.com/jamesmichael/comments
https://qiita.com/Inflatablecanopytents/comments
https://qiita.com/TentDepot01/comments
https://qiita.com/starlinetentsgeorgia/comments

**Reply by kskwtnk** [reply]
@yoshinoyuya 
ご意見ありがとうございます！

我々も認識しているのですが、対応が追いついていない箇所でもあります…… 🙇 
リンクを貼っていただいた分はスパムであったため、対応完了いたしました 🙏 

スパムの手口も巧妙化しており、今までの対策フローをすり抜けてくるものが多くなってきています
そのためこういった報告や通報などをしていただけるのは非常にありがたいです！！！！

なかなか根本的な解決ができずご迷惑をおかけしますが今後ともよろしくお願いします 🙇

**Reply by yoshinoyuya** [reply]
@xrxoxcxox 
迅速な対応ありがとうございます。
今後も何か気付きがありましたら、報告や通報等を行っていきたいと思います。
引き続き、対応よろしくお願いします。
